src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -0,0 +1,172 @@
+    /// </summary>
+    public class DiagnosticsClient
+    {
+        private int _processId;
+        private static string DiagnosticsPortPattern { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? @"^dotnet-diagnostic-(\d+)$" : @"^dotnet-diagnostic-(\d+)-(\d+)-socket$";
+        private static string IpcRootPath { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? @"\\.\pipe\" : Path.GetTempPath();
+
+        public DiagnosticsClient(int processId)
+        {
+            _processId = processId;
+        }
+
+        /// <summary>
+        /// Start tracing the application via CollectTracing2 command.
+        /// </summary>
+        /// <param name="providers">An IEnumerable containing the list of Providers to turn on.</param>
+        /// <param name="requestRundown">If true, request rundown events from the runtime</param>
+        /// <param name="circularBufferMB">The size of the runtime's buffer for collecting events in MB</param>
+        /// <returns>
+        /// An EventPipeSession object representing the EventPipe session that just started.
+        /// </returns> 
+        public EventPipeSession StartEventPipeSession(IEnumerable<EventPipeProvider> providers, bool requestRundown=true, int circularBufferMB=256)
+        {
+            return new EventPipeSession(_processId, providers, requestRundown, circularBufferMB);
+        }
+
+        /// <summary>
+        /// Trigger a core dump generation.
+        /// </summary> 
+        /// <param name="dumpType">Type of the dump to be generated</param>
+        /// <param name="dumpPath">Full path to the dump to be generated. By default it is /tmp/coredump.{pid}</param>
+        /// <param name="logDumpGeneration">When set to true, display the dump generation debug log to the console.</param>
+        public void WriteDump(DumpType dumpType, string dumpPath, bool logDumpGeneration=false)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                throw new PlatformNotSupportedException($"Unsupported operating system: {RuntimeInformation.OSDescription}");
+
+            if (string.IsNullOrEmpty(dumpPath))
+                throw new ArgumentNullException($"{nameof(dumpPath)} required");
+
+            var payload = SerializeCoreDump(dumpPath, dumpType, logDumpGeneration);
+            var message = new IpcMessage(DiagnosticsServerCommandSet.Dump, (byte)DumpCommandId.GenerateCoreDump, payload);
+            var response = IpcClient.SendMessage(_processId, message);
+            var hr = 0;
+            switch ((DiagnosticsServerCommandId)response.Header.CommandId)
+            {
+                case DiagnosticsServerCommandId.Error:
+                    throw new ServerErrorException($"Writing dump failed (HRESULT: 0x{hr:X8})");
+                case DiagnosticsServerCommandId.OK:
+                    hr = BitConverter.ToInt32(response.Payload, 0);
```

Comment:
It looks like you are ignoring the result returned from the client process. It either needs to be returned or an exception thrown if it != S_OK.  And dotnet-dump collect needs to catch whatever exception is thrown.

Reply:
Looking at the code some more it looks like you have the this BitConverter line in the wrong case statement. Shouldn't it be in the Error case?

Reply:
Yup, it should be. Fixed now.